### PR TITLE
Test/content moderation 3836

### LINKED ISSUE #3836: test : add unit tests for content moderation utility
Summary of What Needs to be Done:
The content moderation utility in backend/src/utils/contentModeration.ts checks user-supplied text for CSAM, hate speech, violence, and self-harm patterns. It currently has no test coverage, which leaves critical safety patterns unchecked.

Changes that Need to be Made:
- backend/src/utils/__tests__/contentModeration.test.ts: Create a new jest test file with tests for:
  - checkContent returns { blocked: false } for safe text
  - checkContent returns { blocked: true, reason } for CSAM patterns
  - checkContent returns { blocked: true, reason } for hate speech patterns
  - checkContent returns { blocked: true, reason } for violence patterns
  - checkContent returns { blocked: true, reason } for self-harm patterns
  - assertContentSafe throws for blocked content, does not throw for safe text
  - Empty string returns { blocked: false }

Impact that it would Provide:
- Ensures harmful content patterns are correctly detected
- Guards against pattern regressions if regexes are modified
- Critical safety infrastructure for AI content generation

Note: Please assign this issue to the `tmdeveloper007` account.

### PR DESCRIPTION
**Pull request by vaishnavi003-svg**
N/A - Test file addition.

Added unit tests for content moderation utility (`backend/src/utils/contentModeration.ts`):
- Tests for CSAM, hate speech, violence, self-harm patterns
- Tests for safe text, empty string, edge cases
- `assertContentSafe()` throws for blocked content

- [ ] Bug fix

Fixes #3836

N/A

- [x] I have added screenshots or a recording when they help explain this PR, or noted N/A with a one-line reason.
- [x] I have filled in the related issue number when there is one.
- [x] I have tested my changes.
- [x] I have done a self-review of the code.

### CODE REVIEW

**Comment by vaishnavi003-svg** [comment]
@ronisarkarexe 
please review
thank you

File: backend/src/utils/contentModeration.ts [hunk 1]
```diff
@@ -65,14 +123,26 @@
...
+    throw new Error(`Content blocked: ${result.reason}`);
+  }
+}
+
     throw new Error(
       `Content moderation: your request was blocked because it contains ${result.reason}. Please revise your content.`
     );
   }
 }
+main
```
[the remark this answers is not in the document]
Comment (vaishnavi003-svg) [reply backend/src/utils/contentModeration.ts:148]: resolve

File: frontend/src/utils/storyParser.ts [hunk 2]
```diff
@@ -27,82 +27,103 @@
...
   "Get", "Got", "Let", "Make", "Made", "Said", "Told", "Came", "Went",
-  // Common adjectives / adverbs that appear capitalised mid-sentence
   "Every", "Some", "Many", "Much", "More", "Most", "Such", "Only",
   "Very", "Too", "All", "Both", "Each", "Few", "Own", "Same", "Other",
-  // Location words (capitalised in text sometimes)
   ...LOCATION_WORDS.map(w => w.charAt(0).toUpperCase() + w.slice(1)),
-  // Story-specific common caps
   "Dragons", "Dragon", "Night", "Day", "Morning", "Evening",
+  // ✅ Helper to find all occurrences of a word
+const getAllOccurrences = (content: string, word: string): number[] => {
```
[the remark this answers is not in the document]
Comment (vaishnavi003-svg) [reply frontend/src/utils/storyParser.ts:51]: resolve

**Comment by vaishnavi003-svg** [comment]
@ronisarkarexe
please review
thank you

**Comment by vaishnavi003-svg** [comment]
@ronisarkarexe
conflict resolve please review
thank you

**Comment by vaishnavi003-svg** [comment]
@ronisarkarexe
please review
thank you
